Classes/Controller/GetFileController.php

```diff
@@ -182,6 +182,33 @@ public function attachmentAction()
                 return 'No such action';
         }
 
+        // stop here, if inactive Fedora objects are not allowed to be disseminated
+
+        $restrictToActiveDocuments = TRUE;
+
+        if ('yes' == $piVars['deliverInactive']) {
+            $restrictToActiveDocuments = FALSE;
+        }
+
+        if (TRUE === $restrictToActiveDocuments) {
+            $objectProfileURI = rtrim('http://' . $fedoraHost,"/").'/fedora/objects/'.$piVars['qid'].'?format=XML';
+            $objectProfileXML = file_get_contents($objectProfileURI);
+            if (FALSE !== $objectProfileXML) {
```

Comment:
There could be a problem here. If the Fedora object XML cannot be loaded for some reason, we cannot make an informed decision about whether to disseminate the object or not. In case of failure, the dissemination should be stopped with 500 Server Error or similar.